### PR TITLE
#41 Adding refresh count action

### PR DESCRIPTION
**Pull request by Alemkin**

### CODE REVIEW

File: src/containers/Headers.js [hunk 1]
```diff
@@ -27,7 +28,11 @@ const handlers = {
   updateSort: props => event => {
     const isAsc = !hasClass('sorted-asc', event.target.className)
     const id = event.target.getAttribute('data-id')
-    props.refreshData(props.name, {...props.filter, sorts: updateFilterSort(id, isAsc, props.filter.sorts)})
+    const updatedFilter = {...props.filter, sorts: updateFilterSort(id, isAsc, props.filter.sorts)}
+    if (updatedFilter && !updatedFilter.page) {
```
Comment (frodare) [inline src/containers/Headers.js]: why is there an if condition for count but not data?

File: src/containers/Headers.js [hunk 2]
```diff
@@ -27,7 +28,11 @@ const handlers = {
...
+      props.refreshCount(props.name, updatedFilter)
+    }
+    props.refreshData(props.name, updatedFilter)
```
Comment (frodare) [inline src/containers/Headers.js]: We aren't careful this will result in two XHR calls.

File: src/containers/Manifest.js [hunk 3]
```diff
@@ -36,6 +37,9 @@ const lifecycleMethods = {
     if (this.props.inMemoryData) {
       this.props.setInMemoryData(this.props.name, this.props.inMemoryData)
     } else {
+      if (this.props.filter && !this.props.filter.page) {
```
Comment (frodare) [inline src/containers/Manifest.js]: why is there an if condition for count but not data?

File: src/containers/PageSizer.js [hunk 4]
```diff
@@ -15,13 +15,16 @@ const mapStateToProps = (state, props) => {
...
-  refreshData: actions.refreshData
+  refreshData: actions.refreshData,
+  refreshCount: actions.refreshCount
 }, dispatch)
 
 const handlers = {
   changePageSize: props => event => {
     const pageSize = event.target.value
-    props.refreshData(props.name, {...props.filter, page: 0, pageSize: pageSize})
+    const updatedFilter = {...props.filter, page: 0, pageSize: pageSize}
```
Comment (frodare) [inline src/containers/PageSizer.js]: `const updatedFilter = {...props.filter, page: 0, pageSize}`

File: src/containers/Pager.js [hunk 5]
```diff
@@ -18,13 +18,18 @@ const mapStateToProps = (state, props) => {
...
-  refreshData: actions.refreshData
+  refreshData: actions.refreshData,
+  refreshCount: actions.refreshCount
 }, dispatch)
 
 const handlers = {
   changePage: props => event => {
-    const nextPage = event.target.getAttribute('data-page')
-    props.refreshData(props.name, {...props.filter, page: nextPage})
+    const nextPage = window.parseInt(event.target.getAttribute('data-page'))
```
Comment (frodare) [inline src/containers/Pager.js]: inlude radix:
`const nextPage = window.parseInt(event.target.getAttribute('data-page'), 10)`

File: src/containers/Pager.js [hunk 6]
```diff
@@ -18,13 +18,18 @@ const mapStateToProps = (state, props) => {
...
+    const updatedFilter = {...props.filter, page: nextPage}
+    if (updatedFilter && !updatedFilter.page) {
```
Comment (frodare) [inline src/containers/Pager.js]: why is there an if condition for count but not data?

**Comment by frodare** [comment]
The example is not working in this PR

File: src/containers/Pager.js (hunk 6 above)
Comment (Alemkin) [reply src/containers/Pager.js]: it needs to be there

**Comment by Alemkin** [comment]
example works for me

File: src/containers/Manifest.js (hunk 3 above)
Comment (Alemkin) [reply src/containers/Manifest.js]: needs to be there

File: src/containers/Headers.js (hunk 2 above)
Comment (Alemkin) [reply src/containers/Headers.js]: not sure what you mean

File: src/containers/Headers.js (hunk 1 above)
Comment (Alemkin) [reply src/containers/Headers.js]: needs to be there

**Pull request merged**